Tests/Imagine/Filter/PostProcessor/MozJpegPostProcessorTest.php

```diff
@@ -11,25 +11,113 @@
+     */
+    public function testDeprecatedSetQualityMethod(): void
     {
-        $mozJpegPostProcessor = new MozJpegPostProcessor(
-            __DIR__.'/../../../Fixtures/bash/empty-command.sh'
-        );
+        $this->getPostProcessorInstance()->setQuality(50);
+    }
+
+    public static function provideProcessArgumentsData(): array
+    {
+        $data = [
+            [[], ['-quant-table', 2, '-optimise']],
+            [['quant_table' => 10], ['-quant-table', 10, '-optimise']],
+            [['optimise' => false], ['-quant-table', 2]],
+            [['optimise' => true], ['-quant-table', 2, '-optimise']],
+            [['quality' => 50], ['-quant-table', 2, '-optimise', '-quality', 50]],
+            [['quant_table' => 4, 'optimise' => true, 'quality' => 100], ['-quant-table', 4, '-optimise', '-quality', 100]],
+        ];
+
+        return array_map(function (array $d) {
+            array_unshift($d[1], AbstractPostProcessorTestCase::getPostProcessAsStdInExecutable());
+
+            return $d;
+        }, $data);
+    }
+
+    /**
+     * @dataProvider provideProcessArgumentsData
+     */
+    public function testProcessArguments(array $options, array $expected): void
+    {
+        $this->assertSame($expected, $this->getProcessArguments($options));
+    }
+
+    public static function provideProcessData(): array
+    {
+        $file = 'stdio-file-content-string';
+        $data = [
+            [[], '-quant-table 2 -optimise'],
+            [['quant_table' => 10], '-quant-table 10 -optimise'],
+            [['optimise' => false], '-quant-table 2'],
+            [['optimise' => true], '-quant-table 2 -optimise'],
+            [['quality' => 50], '-quant-table 2 -optimise -quality 50'],
+            [['quant_table' => 4, 'optimise' => true, 'quality' => 100], '-quant-table 4 -optimise -quality 100'],
+        ];
+
+        return array_map(function ($d) use ($file) {
+            array_unshift($d, $file);
+
+            return $d;
+        }, $data);
+    }
 
-        $binary = new Binary('content', 'image/jpg', 'jpg');
-        $result = $mozJpegPostProcessor->process($binary);
+    /**
+     * @dataProvider provideProcessData
+     */
+    public function testProcess(string $content, array $options, string $expected): void
+    {
+        $file = sys_get_temp_dir().'/test.jpeg';
+        file_put_contents($file, $content);
+
+        $process = $this->getPostProcessorInstance();
+        $result = $process->process(new FileBinary($file, 'image/jpeg', 'jpeg'), $options);
+
+        $this->assertContains($expected, $result->getContent());
+        $this->assertContains($content, $result->getContent());
+
+        @unlink($file);
+    }
+
+    /**
+     * @dataProvider provideProcessData
+     *
+     * @expectedException \Symfony\Component\Process\Exception\ProcessFailedException
```

Comment:
this is deprecated, the recommended way is to do `$this->expectException(ProcessFailedException::class);` on the line right before the one that should trigger the exception.